api/jsonapi.go

```diff
@@ -2223,6 +2224,92 @@ func (i *jsonAPIHandler) POSTReleaseEscrow(w http.ResponseWriter, r *http.Reques
 	SanitizedResponse(w, `{}`)
 }
 
+func (i *jsonAPIHandler) POSTSignMessage(w http.ResponseWriter, r *http.Request) {
+	type plaintext struct {
+		Content string `json:"content"`
+	}
+	var msg plaintext
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&msg)
+	if err != nil {
+		ErrorResponse(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	sig, err := i.node.IpfsNode.PrivateKey.Sign([]byte(msg.Content))
+	if err != nil {
+		ErrorResponse(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	keyBytes, err := i.node.IpfsNode.PrivateKey.Bytes()
+	if err != nil {
+		ErrorResponse(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	SanitizedResponse(w, fmt.Sprintf(`{"signature": "%s","pubkey":"%s","peerId":"%s"}`,
+		hex.EncodeToString(sig),
+		hex.EncodeToString(keyBytes),
```

Comment:
This is returning the private key and not the public key.

Reply:
Well it seems fix but unfortunately it's not converting the privkey to pubkey properly.